sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestVariousOddities.java

```diff
@@ -151,4 +154,49 @@ public void testAbiguityResolved() {
         @Override
         String value();
     }
+
+    public interface OnDemandOddities extends SqlObject, Transactional<OnDemandOddities> {
+        @CreateSqlObject
+        VersionA versionA();
+    }
+
+    public class DecoratedOnDemandOddities implements OnDemandOddities {
+        private OnDemandOddities onDemand = dbRule.getJdbi().onDemand(OnDemandOddities.class);
+
+        @Override
+        public Handle getHandle() {
+            return onDemand.getHandle();
+        }
+
+        @Override
+        public <R, X extends Exception> R withHandle(HandleCallback<R, X> callback) throws X {
+            return onDemand.withHandle(callback);
+        }
+
+        @Override
+        public VersionA versionA() {
+            return onDemand.versionA();
+        }
+    }
+
+    @Test
+    public void onDemandCreateSqlObject() throws Exception {
+        assertThat(dbRule.getJdbi().onDemand(OnDemandOddities.class).versionA().value())
+                .isEqualTo("intriguing");
+    }
+
+    @Test
+    public void decoratedOnDemandWithHandleInTransaction() throws Exception {
+        OnDemandOddities onDemand = new DecoratedOnDemandOddities();
+        onDemand.withHandle(h ->
+            h.inTransaction(txn ->
+                assertThat(onDemand.versionA().value()).isEqualTo("intriguing")));
+    }
+
+    @Test
+    public void decoratedOnDemandInTransaction() throws Exception {
+        OnDemandOddities onDemand = new DecoratedOnDemandOddities();
+        onDemand.inTransaction(txn ->
+                assertThat(onDemand.versionA().value()).isEqualTo("intriguing"));
```

Comment:
So this fails because onDemand opens and closes a connection around method invocations, but this `versionA()` returns the created object, which of course cannot be called outside the onDemand method because its handle is closed upon returning.

Reply:
And the idea for the fix is that since this invocation happens inside the scope of the same onDemand object's `Transactional<>` method, it should share the outer Handle instead of opening+closing a redundant onDemand one.

Reply:
Exactly!  I think it's a `LazyHandleSupplier` that ends up getting shared, but that's the right idea.